File: GraphleManager/src/main/java/com/graphle/graphlemanager/Tag/Tag.java
package com.graphle.graphlemanager.Tag;

import org.springframework.data.neo4j.core.schema.GeneratedValue;
import org.springframework.data.neo4j.core.schema.Id;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import java.util.UUID;

public record Tag(@Id @GeneratedValue UUID id, @NonNull String name, @Nullable String value, @Nullable Double numericValue) {
    public Tag(@NonNull String name, @Nullable String value, @Nullable Double numericValue) {
        this(null, name, value, numericValue);
    }
}

File: GraphleManager/src/main/kotlin/com/graphle/graphlemanager/Tag/Tag.kt
package com.graphle.graphlemanager.Tag

import org.springframework.data.neo4j.core.schema.GeneratedValue
import org.springframework.data.neo4j.core.schema.Id
import org.springframework.lang.NonNull
import org.springframework.lang.Nullable
import java.util.*

@JvmRecord
data class Tag(
    @field:GeneratedValue @field:Id val id: UUID?,
    @field:NonNull @param:NonNull val name: String,
    @field:Nullable @param:Nullable val value: String?,
    @field:Nullable @param:Nullable val numericValue: Double?
) {
    constructor(@NonNull name: String, @Nullable value: String?, @Nullable numericValue: Double?) : this(
        null,
        name,
        value,
        numericValue
    )
}
